### PR TITLE
Remove qubits from `TomographyExperiment`

### PR DESCRIPTION
**Pull request by msohaibalam**
Necessary update after merging of https://github.com/rigetti/pyquil/pull/896

### CODE REVIEW

**Comment by msohaibalam** [comment]
@astaley @jonward-rigetti @kylegulshen New PR for removal of `qubits` from `TomographyExperiment`

**Comment by blakejohnson** [comment]
Per the discussion in https://github.com/rigetti/pyquil/pull/896, I would argue the correct solution is to add the `qubits` argument back order to not break semver expectations.

File: requirements.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-pyquil>=2.4.0
+pyquil>=2.7.0
```
Comment (blakejohnson) [inline requirements.txt]: This change must imply a major version bump. We need to either revert the change to pyquil and re-release, or re-label this release as 3.0.0.

**Comment by blakejohnson** [comment]
I'm ok proceeding here now that we have a plan in place to provide a deprecation path on the prior API in pyquil.

**Comment by msohaibalam** [comment]
Sounds good, merging this in now.

**Comment by blakejohnson** [comment]
@msohaibalam: @astaley found that `acquire_dfe_data` still makes reference to the `qubits` attribute of `TomographyExperiment`. This points out that we haven't quite achieved full backwards compatibility in pyquil 2.7.1. We still need to populate the `qubits` attribute with the value that it would have had previously.

**Comment by msohaibalam** [comment]
I see this particular line: https://github.com/rigetti/forest-benchmarking/blob/master/forest/benchmarking/direct_fidelity_estimation.py#L324.

Would it work if we replaced `expr.qubits` there with the qubits appearing in the estimated `PauliTerm`? From the description of `estimate_dfe`, it sounds like `dimension` is referring to the dimension of the Hilbert space that the measured `PauliTerm` is acting on.

File: forest/benchmarking/direct_fidelity_estimation.py [hunk 2]
```diff
@@ -318,9 +318,9 @@ def acquire_dfe_data(qc: QuantumComputer, expr: TomographyExperiment, n_shots=10
                    program=expr.program,
                    out_pauli=[str(e[0].out_operator) for e in expr],
                    pauli_point_est=np.array([r.expectation for r in res]),
-                   pauli_std_err=np.array([r.stddev for r in res]),
+                   pauli_std_err=np.array([r.std_err for r in res]),
                    cal_point_est=np.array([r.calibration_expectation for r in res]),
-                   cal_std_err=np.array([r.calibration_stddev for r in res]),
+                   cal_std_err=np.array([r.calibration_std_err for r in res]),
                    dimension=2**len(expr.qubits),
                    qubits=expr.qubits)
```
Comment (blakejohnson) [inline forest/benchmarking/direct_fidelity_estimation.py]: These two lines refer to `expr.qubits`.

**Comment by blakejohnson** [comment]
> Would it work if we replaced `expr.qubits` there with the qubits appearing in the estimated `PauliTerm`? From the description of `estimate_dfe`, it sounds like `dimension` is referring to the dimension of the Hilbert space that the measured `PauliTerm` is acting on.

Yes, that would work. We also need to populate the `qubits` field of `DFEData`.

**Comment by msohaibalam** [comment]
@blakejohnson Thanks for the input! I'm working on PR #115 to fix this bug asap. I'll ping you there once I'm sure the tests pass.
